motoman_driver/MotoPlus/MotionServer.c

```diff
@@ -299,34 +302,44 @@ void Ros_MotionServer_WaitForSimpleMsg(Controller* controller, int connectionInd
 				default:
 					bInvalidMsgType = TRUE;
 					break;
-			}
-		}
-		
-		// Check message size
-		if(byteSize >= expectedSize)
-		{
-			// Process the simple message
-			ret = Ros_MotionServer_SimpleMsgProcess(controller, &receiveMsg, byteSize, &replyMsg);
-			if(ret == 1)
+        	}
+        }
+
+        bHasPreviousData = FALSE;
+        // Check message size
+       	if(byteSize >= expectedSize && expectedSize <= sizeof(SimpleMsg))
+       	{
+       		// Process the simple message
+          	ret = Ros_MotionServer_SimpleMsgProcess(controller, &receiveMsg, expectedSize, &replyMsg);
+			if(ret == 1) 
+			{
 				bDisconnect = TRUE;
+			}
+			else if( byteSize > expectedSize ) 
+			{
+				printf("MessageReceived(%d bytes): expectedSize=%d, processing rest of bytes (%d, %d, %d)\r\n", byteSize,  expectedSize, sizeof(receiveMsg), receiveMsg.body.jointTrajData.sequence, ((int*)((char*)&receiveMsg +  expectedSize))[5]);
+				memmove(&receiveMsg, (char*)&receiveMsg + expectedSize, byteSize-expectedSize);
+				byteSize -= expectedSize;
+				bHasPreviousData = TRUE;
+			}

```

Comment:
These lines now use spaces for indentation, where the rest of the file uses tabs.